stub/src/test/java/io/grpc/stub/ClientCallsTest.java

```diff
@@ -503,6 +504,50 @@ public void onCompleted() {
     assertThat(requests).isEmpty();
   }
 
+  @Test
+  public void checkForNullInAsyncUnaryCall()  {
+    NoopClientCall<Integer, String> call = new NoopClientCall<>();
+    try {
+      ClientCalls.asyncUnaryCall(call, Integer.valueOf(1), null);
+      Assert.fail("Should have gotten an exception for the null responseObserver");
+    } catch (NullPointerException e) {
+      assertSame("responseObserver", e.getMessage());
```

Comment:
Why not `assertEquals`? Do we care that the instance of the string is the same as returned by error message?

Reply:
Good point.  It was being used all over the test so I've updated all of the assertSame to assertEquals.